pulp_2to3_migration/tests/functional/test_migration_behaviour.py

```diff
@@ -0,0 +1,337 @@
+import json
+import time
+import unittest
+
+from pulpcore.client.pulpcore import Configuration
+
+from pulpcore.client.pulp_file import (
+    ApiClient as FileApiClient,
+    ContentFilesApi,
+    DistributionsFileApi,
+    PublicationsFileApi,
+    RemotesFileApi,
+    RepositoriesFileApi,
+    RepositoriesFileVersionsApi
+)
+from pulpcore.client.pulp_2to3_migration import (
+    ApiClient as MigrationApiClient,
+    MigrationPlansApi,
+    Pulp2ContentApi,
+    Pulp2RepositoriesApi
+)
+
+from pulp_2to3_migration.tests.functional.util import get_psql_smash_cmd, set_pulp2_snapshot
+
+from pulp_smash import cli
+from pulp_smash import config as smash_config
+from pulp_smash.pulp3.bindings import monitor_task, monitor_task_group
+
+from .common_plans import FILE_SIMPLE_PLAN, FILE_COMPLEX_PLAN
+from .constants import BINDINGS_CONFIGURATION, TRUNCATE_TABLES_QUERY_BASH
+
+
+MANY_INTO_ONE_PLAN = json.dumps({
+    "plugins": [{
+        "type": "iso",
+        "repositories": [
+            {
+                "name": "file",
+                "pulp2_importer_repository_id": "file",  # policy: immediate
+                "repository_versions": [
+                    {
+                        "pulp2_repository_id": "file"  # content count: iso - 3
+                    },
+                    {
+                        "pulp2_repository_id": "file-large"  # content count: iso - 10
+                    },
+                    {
+                        "pulp2_repository_id": "file-many"  # content count: iso - 250
```

Comment:
file-many is on_demand repo and it's content won't be migrated unless you specify pulp2_importer_id: file-many, that's why line 259 fails since it does not contain 250 units.
However besides this observation it indeed seems that there is an issue in the logic.
New repo_version discards the content from previous repo_version completely, meaning that newly created version does not have 'additive' behaviour https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/app/migration.py#L369
My expectation is to see additive behaviour, is it correct or i am missing something?

Reply:
Agreed with expectation around the on_demand content.

I had the same thinking process and then thinking more about it decided that additive wouldn't make sense in the migration case. It means no content will be ever removed in the later versions.
 Katello confirmed that they need a 'mirror' behaviour, since pulp repo versions will correspond to CV whichin pulp2 had to be in an individual repo each.

I'll adjust the test to reflect this behaviour.

Thanks for feedback